akka-docs/src/main/paradox/cluster-singleton.md

```diff
@@ -156,4 +156,23 @@ or create it from another config section with the same layout as below. `Cluster
 a parameter to the `ClusterSingletonProxy.props` factory method, i.e. each singleton proxy can be configured
 with different settings if needed.
 
-@@snip [reference.conf]($akka$/akka-cluster-tools/src/main/resources/reference.conf) { #singleton-proxy-config }
\ No newline at end of file
+@@snip [reference.conf]($akka$/akka-cluster-tools/src/main/resources/reference.conf) { #singleton-proxy-config }
+
+## Supervision
+
+Sometimes it is useful to add supervision for the Cluster Singleton itself. To accomplish this you need to add a parent supervisor actor which will be used to create the 'real' singleton instance. Below is an example implementation (credit to [this StackOverflow answer](https://stackoverflow.com/a/36716708/779513))
+
+Scala
+:  @@snip [ClusterSingletonSupervision.scala]($akka$/akka-docs/src/test/scala/docs/cluster/singleton/ClusterSingletonSupervision.scala) { #singleton-supervisor-actor }
+
+Java
+:  @@snip [SupervisorActor.java]($akka$/akka-docs/src/test/java/jdocs/cluster/singleton/SupervisorActor.java) { #singleton-supervisor-actor }
+
+And used here
+
+Scala
+:  @@snip [ClusterSingletonSupervision.scala]($akka$/akka-docs/src/test/scala/docs/cluster/singleton/ClusterSingletonSupervision.scala) { #singleton-supervisor-actor-usage }
+
+Java
+:  @@snip [ClusterSingletonSupervision.java]($akka$/akka-docs/src/test/java/jdocs/cluster/singleton/ClusterSingletonSupervision.java) { #singleton-supervisor-actor-usage-imports }
+@@snip [ClusterSingletonSupervision.java]($akka$/akka-docs/src/test/java/jdocs/cluster/singleton/ClusterSingletonSupervision.java) { #singleton-supervisor-actor-usage }
```

Comment:
just sanity checking if this renders okey?

Reply:
It seemed to for me, but I'll admit that the full styling wasn't working so I can't be 100%

Reply:
I think I've seen this sometimes work and sometimes not - IIRC a safer way is to put it on one line like `@@snip [ClusterSingletonSupervision.java]($akka$/akka-docs/src/test/java/jdocs/cluster/singleton/ClusterSingletonSupervision.java) { #singleton-supervisor-actor-usage-imports #singleton-supervisor-actor-usage }`

Reply:
Thanks @raboof   I made that adjustment :)

Reply:
This was fine actually; it looked quite good; now the code is not valid Java code and looks weird; you can not import like this in-line in Java. @raboof

Reply:
Oops sorry for the distraction then!

Reply:
This looked good and more "real":
![cluster_singleton_ _akka_documentation](https://user-images.githubusercontent.com/120979/43821921-678fb418-9b25-11e8-92d4-85593c91c2a8.png)

Reply:
No worries, I guess I'll revert the change and merge...

Reply:
jup if that works that seems neat

Reply:
I just reverted manually

Reply:
thanks! :)